src/python/pants/backend/docker/util_rules/docker_binary.py

```diff
@@ -98,22 +121,43 @@ class DockerBinaryRequest:
     search_path: SearchPath = DockerBinary.DEFAULT_SEARCH_PATH
 
 
-@rule(desc="Finding the `docker` binary", level=LogLevel.DEBUG)
+@rule(desc="Finding the `docker` binary and related tooling", level=LogLevel.DEBUG)
 async def find_docker(
     docker_request: DockerBinaryRequest, docker_options: DockerOptions
 ) -> DockerBinary:
     env = await Get(Environment, EnvironmentRequest(["PATH"]))
-    search_path = docker_options.executable_search_path(env)
-    request = BinaryPathRequest(
-        binary_name="docker",
-        search_path=search_path or docker_request.search_path,
-        test=BinaryPathTest(args=["-v"]),
+    search_path = docker_options.executable_search_path(env) or docker_request.search_path
+    requests = [
+        BinaryPathRequest(
+            binary_name=tool,
+            search_path=search_path,
+            test=BinaryPathTest(args=["-v"]) if tool == "docker" else None,
+        )
+        for tool in {"docker", *docker_options.tools}
+    ]
+    binary_paths = await MultiGet(
+        Get(BinaryPaths, BinaryPathRequest, request) for request in requests
+    )
+
+    found_paths = []
+    for binary, request in zip(binary_paths, requests):
+        if binary.first_path:
+            found_paths.append(binary.first_path)
+        else:
+            raise BinaryNotFoundError.from_request(
+                request,
+                rationale="interact with the docker daemon",
+            )
+
+    return DockerBinary(
+        found_paths[0].path,
+        found_paths[0].fingerprint,
+        extra_env=(
+            {"PATH": os.pathsep.join({os.path.dirname(found.path) for found in found_paths[1:]})}
+            if len(found_paths) > 1
+            else None
```

Comment:
I'm a bit concerned with this solution, as we're giving access to *all* executables sharing the same directory as the one we were searching for.

What I'd like is to get the equivalent of what is achieved with this snippet for the `shell_command` target..
https://github.com/pantsbuild/pants/blob/c52d2cd27152a79d00237d88ce0f299c6a944b8a/src/python/pants/backend/shell/shell_command.py#L213-L215

If this is a recurring theme, perhaps some generic feature for isolating a set of binaries is warranted..?

Reply:
Implemented by #14374